test/client.spec.ts

```diff
@@ -85,27 +100,34 @@ describe('Certificate Client', () => {
 
 	// Attempt to generate a certificate and associated local IP address
 	describe('Certificate generation and DNS A record', () => {
-		const testIP = '10.1.2.3';
+		const testIp = '10.1.2.3';
+		const updatedIp = '10.1.2.4';
+		const configRoot = `${process.cwd()}/test/testcert`;
+
+		// Ensure cert directory is cleaned up post-test
+		after(() => {
+			return removeDirectory(configRoot);
+		});
 
 		it('should generate a valid certificate', () => {
 			const client = new BalenaCertificateClient({
 				dnsUpdateHost: dnsHost,
 				dnsUpdatePort: dnsPort,
 				authToken: process.env.CERT_TEST_AUTH_TOKEN || '',
-				configRoot: `${process.cwd()}/test/testcert`,
+				configRoot,
 			});
 
 			return client
 				.requestCertificate({
 					domain,
 					subdomains: [`*.${uuid}`, `*.devices.${uuid}`],
-					ip: '10.1.2.3',
+					ip: testIp,
 					email,
 					renewing: false,
 				})
 				.then(result => {
 					if (!result) {
-						return new Error('result should not be undefined');
+						return new Error('`result` should not be undefined');
```

Comment:
nit: Backticks are used for string interpolation. This threw me off; as I was looking to see what was being interpolated and saw nothing since they are being used as single quotes. I would prefer we just escaped single quote instead of this.

Reply:
Backticks with no quoting are indeed for template strings. However, there's no reason not to use them in an output string. ;-)

Reply:
Did you ever write C documentation @hedss ? :P